### DISCUSSION TITLE
How can I see stats of my GRPC server using statsviz?

### DISCUSSION THREAD

**Discussion by arl** [arl/statsviz#70] [Q&A] [answered]
Question initially asked by asked by @saltperfect in issue #69 (which i moved to the *discussions* tab):

> If statsviz cannot help is there a alternative?

**Comment by arl** [comment, marked as answer]
@saltperfect

statsviz serves a web gui via http that shows real time statistics of the Go runtime (garbage collector, etc.) of your application. It's made to monitor in real time the allocations pattern and memory usage of your app. As such, it can do so with any kind of Go application.

However if you're looking at exposing application-specific statistics such as the number of GRPC calls per second for example, statsviz has no clue about that and has currently no way to add custom metrics. It's planned but there's no ETA yet (see #62 )

**Comment by saltperfect** [comment]
@arl 
When I used `mux := http.NewServeMux() statsviz.Register(mux)` I was getting 404 but when I used `statsviz.RegisterDefault()`
I was able to see stats on my browser. Can you guess what could be the reason?

**Reply by arl** [reply]
Hard to say without seeing the code. It all depends on what the rest of the code does with `mux`.
Did you have a look at [_example/mux](https://github.com/arl/statsviz/blob/main/_example/mux/main.go)?

**Reply by saltperfect** [reply]
Rest of the mux doesn't do anything. I have a grpc server. I think that might be the reason.

**Reply by arl** [reply]
Did you manage to solve your issue?
